### DISCUSSION TITLE
Eager loaded relations (defined in $with array) uses softDeleted scope

### DISCUSSION THREAD

**Discussion by ripplexyz** [laravel/nova-issues#6243] [Ideas & Feature Requests]
- Laravel Version: 10.29
- Nova Version: 4.28
- PHP Version: 8.2
- Database Driver & Version: MySQL 8+
- Operating System and Version: MacOS 13.6
- Browser type and version: Chrome 118.0.5993.117
- Reproduction Repository: https://github.com/###/###

### Description:

$with relationship models are loaded with soft delete filters.

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create Resource A with belongsTo relationship to SoftDeletable Resource B.
2. Create Resource B row -> lets call it B1
3. Create Resource A row -> lets call it A1 which is belong to B1
4. Soft Delete Resource B1
5. For Resource A, eager load Resource B relationship using $with = ['b']

Issue: While checking list of Resource A, we can see that the 'b' column is empty since B1 is soft deleted. The issue doesn't happen if eager loading not used.

Expected: While using eager loading with BelongsTo field, skip soft deleted scope or use `$displaysWithTrashed`.

**Comment by crynobone** [comment]
From my understanding, it's actually lazy loading is having an incorrect behavior at the moment. Will need to discuss this more internally to ensure the correct behavior is used consistently.
